cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -70,19 +72,24 @@ class BufferExpander
     }
 
     BufferExpander(std::shared_ptr<ConcreteBuffer> const & buf, bool clone, ctor_passkey const &)
-        : m_concrete_buffer(clone ? buf->clone() : buf)
-        , m_begin(m_concrete_buffer->data())
-        , m_end(m_begin + m_concrete_buffer->size())
-        , m_end_cap(m_begin + m_concrete_buffer->size())
+        : BufferBase<BufferExpander>() // don't delegate m_begin and m_end, which will be overwritten later
+        , m_concrete_buffer(clone ? buf->clone() : buf)
     {
+        m_begin = m_concrete_buffer->data(); // override m_begin and m_end once we have the data
+        m_end = m_begin + m_concrete_buffer->size();
+        m_end_cap = m_begin + m_concrete_buffer->size();
     }
 
     BufferExpander(size_type nbyte, ctor_passkey const &)
+        : BufferBase<BufferExpander>() // don't delegate m_begin and m_end, which will be overwritten later
     {
         expand(nbyte);
     }
 
-    BufferExpander(ctor_passkey const &) {}
+    BufferExpander(ctor_passkey const &)
+        : BufferBase<BufferExpander>() // don't delegate m_begin and m_end, which will be overwritten later
```

Comment:
I think this needs `m_begin` and `m_end` to be initialized.

Reply:
no, `expand()` calls `reserve()`, which initialized `m_begin` and `m_end`.

Reply:
sure, fixed the logic.